lib/Modules/HyperVNetworking/HyperVNetworking.psm1

```diff
@@ -0,0 +1,193 @@
+# Copyright 2014-2016 Cloudbase Solutions Srl
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+Import-Module JujuHooks
+Import-Module JujuWindowsUtils
+Import-Module OpenStackCommon
+Import-Module JujuLogging
+Import-Module Networking
+
+
+function Get-NetType {
+    $cfg = Get-JujuCharmConfig
+
+    if($cfg['network-type'] -notin $NOVA_VALID_NETWORK_TYPES) {
+        Throw ("Invalid network type: '{0}'" -f @($cfg['network-type']))
+    }
+
+    if((Get-IsNanoServer) -and ($cfg['network-type'] -ne 'hyperv')) {
+        Throw ("'{0}' network type is not supported on Nano Server" -f @($cfg['network-type']))
+    }
+
+    return $cfg['network-type']
+}
+
+function Get-NICsByMAC {
+    Param(
+        [Parameter(Mandatory=$false)]
+        [string[]]$MACAddresses
+    )
+
+    if (!$MACAddresses.Count) {
+        return ,@()
+    }
+
+    [System.Array]$nics = Get-NetAdapter | Where-Object {
+        $_.MacAddress -in $MACAddresses -and
+        $_.DriverFileName -notin @("vmswitch.sys", "NdisImPlatform.sys")
+    }
+
+    if(!$nics) {
+        return ,@()
+    }
+
+    return ,$nics
+}
+
+function Get-NICsByName {
+    Param(
+        [Parameter(Mandatory=$false)]
+        [string[]]$Names
+    )
+
+    if (!$Names.Count) {
+        return ,@()
+    }
+
+    [System.Array]$nics = Get-NetAdapter | Where-Object {
+        $_.Name -in $Names -and
+        $_.DriverFileName -ne "vmswitch.sys"
+    }
+
+    if(!$nics) {
+        return ,@()
+    }
+
+    return ,$nics
```

Comment:
The comma stops the default PowerShell behavior of altering function return values (ex: if array has one element, it returns the element, instead of an array with 1 element).

But I agree with this as well. I will change it.